src/libstore/outputs-spec.hh

```diff
@@ -63,6 +67,9 @@ struct OutputsSpec : _OutputsSpecRaw {
     static OutputsSpec parse(std::string_view s);
     static std::optional<OutputsSpec> parseOpt(std::string_view s);
 
+    /* Filter a DrvOutputs by these output names */
+    DrvOutputs filter(DrvOutputs&& input) const;
```

Comment:
I think this would be better as a method on `DrvOutputs` than `OutputsSpec`. `OutputsSpec` is the much simpler data type that shouldn't know anything about more complex things, even if the header just gets an abstract type.

Reply:
`DrvOutputs` is a `std::map`. Should I subclass or just write a separate function?

Reply:
@yorickvP Ah, I see. Either a standalone function or make a struct with one field then, I think.